watchgod/watcher.py

```diff
@@ -25,19 +25,29 @@ def should_watch_dir(self, entry):
     def should_watch_file(self, entry):
         return True
 
-    def _walk(self, dir_path, changes, new_files):
+    def _walk(self, path, changes, new_files):
+        if os.path.isfile(path):
+            self._watch_file(path, changes, new_files)
+        else:
+            self._walk_dir(path, changes, new_files)
+
+    def _watch_file(self, path, changes, new_files, mtime=None):
+        if mtime is None:
+            mtime = os.stat(path).st_mtime
+        new_files[path] = mtime
+        old_mtime = self.files.get(path)
+        if not old_mtime:
+            changes.add((Change.added, path))
+        elif old_mtime != mtime:
+            changes.add((Change.modified, path))
+
+    def _walk_dir(self, dir_path, changes, new_files):
         for entry in os.scandir(dir_path):
-            if entry.is_dir():
-                if self.should_watch_dir(entry):
-                    self._walk(entry.path, changes, new_files)
-            elif self.should_watch_file(entry):
+            if entry.is_dir() and self.should_watch_dir(entry):
```

Comment:
I think revert to the double if we had before, that way we don't call `elif entry.is_file() and self.should_watch_file(entry):` in the case that `entry` is a directory but we decide not to watch it.